MAVProxy/mavproxy.py

```diff
@@ -523,7 +523,7 @@ def process_master(m):
         sys.stdout.flush()
         return
 
-    if m.first_byte and opts.auto_protocol:
+    if m.first_byte and (opts.mavversion == "" or opts.auto_protocol):
```

Comment:
Should use `is None`, not `== ""`; if no argument is given to MAVProxy that's the value it will have here.